===================================================================
RCS file: /cvs/gnome/gtk+/gtk/gtkfilechooserbutton.c,v
retrieving revision 1.52.2.6
retrieving revision 1.52.2.7
diff -u -r1.52.2.6 -r1.52.2.7
--- gtk+/gtk/gtkfilechooserbutton.c	2006/09/22 17:48:24	1.52.2.6
+++ gtk+/gtk/gtkfilechooserbutton.c	2006/09/30 13:27:15	1.52.2.7
@@ -2207,10 +2207,13 @@
 	    GtkFilePath *base_path;
 
 	    base_path = gtk_file_system_volume_get_base_path (priv->fs, data);
-	    row_found = (paths &&
-			 paths->data &&
-			 gtk_file_path_compare (base_path, paths->data) == 0);
-	    gtk_file_path_free (base_path);
+            if (base_path)
+              {
+	        row_found = (paths &&
+		  	     paths->data &&
+			     gtk_file_path_compare (base_path, paths->data) == 0);
+	        gtk_file_path_free (base_path);
+              }
 	  }
 	  break;
 	default:
